src/main/java/com/sngular/kloadgen/processor/objectcreatorfactory/impl/ProtobufObjectCreatorFactory.java

```diff
@@ -46,7 +51,12 @@ public ProtobufObjectCreatorFactory(final Object schema, final BaseSchemaMetadat
       this.schema = SchemaProcessorUtils.buildProtoDescriptor((ProtoFileElement) ((ParsedSchema) schema).rawSchema(), metadata);
     } else if (schema instanceof ProtoFileElement) {
       this.schema = SchemaProcessorUtils.buildProtoDescriptor((ProtoFileElement) schema, metadata);
-    } else {
+    } else if (schema instanceof BaseParsedSchema) {
+      BaseParsedSchema schemaParse = (BaseParsedSchema) schema;
+      ParsedSchemaAdapter adapterParse = schemaParse.getParsedSchemaAdapter();
+      final Object schemaParsed = adapterParse.getRawSchema();
+      this.schema = SchemaProcessorUtils.buildProtoDescriptor((ProtoFileElement) schemaParsed, metadata);
+  } else {
```

Comment:
Final variables and unindented else, same as in avro

Reply:
changed